### PR TITLE
run doctests as part of cabal test

### PR DESCRIPTION
**Pull request by marcosh**
Some notes:

- I had to add the `--write-ghc-environment-files=always` option to `cabal test` to please `doctest-parallel` (see https://github.com/martijnbastiaan/doctest-parallel/blob/main/example/README.md#cabalproject). I also had to gitignore the output of that option
- I had to move the `--color` option to the `.hspec` file because otherwise the option would have been passed also to `doctest-parallel` which does not understand it
- Moved `singletons-base` and `text` out of global dependencies since they are not used by the `crm-doctets` stanza
- I found out that you can leave doctests inside multi-line comments

### CODE REVIEW

**Comment by marcosh** [comment]
closes #4

File: src/CRM/Graph.hs [hunk 1]
```diff
@@ -10,10 +10,10 @@ newtype Graph a = Graph [(a, a)]
  It has as vertices the product of the set of vertices of the initial graph.
  It has as edge from `(a1, b1)` to `(a2, b2)` if and only if there is an edge
  from `a1` to `a2` and an edge from `b1` to `b2`
--}
 
--- >>> productGraph (Graph [('a', 'b')]) (Graph [('c', 'd')])
--- Graph [(('a','c'),('b','d'))]
+ >>> productGraph (Graph [('a', 'b')]) (Graph [('c', 'd')])
+ Graph [(('a','c'),('b','d'))]
+-}
```
Comment (AlexeiDrake) [inline src/CRM/Graph.hs:16]: As I recall, doctests in multiline comments don't execute in VSCode (and potentially other HLS-powered editors too). That is unfortunate.

Did having it as multiline docstring + single line doctests cause any issues?

Comment (marcosh) [reply src/CRM/Graph.hs:16]: so, execution in vscode works only with single-line style

doctests are recognised only when inside Haddock comments (i.e. the ones with a `|`).

For example

```haskell
{- | bla bla bla
-}

-- >>> example
-- result
```

would not work (meaning that the example is executable in vscode but won't be recognised by doctests).

I guess the only viable solution to get both working is using single-line Haddock comments

Comment (AlexeiDrake) [reply src/CRM/Graph.hs]: I think that using single-line comments for everything is the best compromise.

File: decision-log/2023-01-12-use-single-line-haddock-comments.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  [Haddock](https://haskell-haddock.readthedocs.io)-style comments. Haddock
+  supports two styles of comments:
+
+  - single-line: `-- |`
+
+  - multi-line: `{- |`
+
+  Moreover we would like to use `doctests` to properly check the examples
+  included in the Haddock documentation. Such examples are defined using the
+  `>>>` syntax inside Haddock comments.
```
Comment (AlexeiDrake) [inline decision-log/2023-01-12-use-single-line-haddock-comments.yaml:17]:
```suggestion
  Haddock also offers "doctests" to automatically run the examples
  included in the Haddock documentation. Such examples are defined using the
  `>>>` syntax inside Haddock comments.
```

File: decision-log/2023-01-12-use-single-line-haddock-comments.yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+  The [Haskell Language Server](https://haskell-language-server.readthedocs.io)
+  allows to execute directly examples defined with the `>>>` syntax inside any
+  kind (i.e. not necessarily Haddock ones) on single-line comments.
```
Comment (AlexeiDrake) [inline decision-log/2023-01-12-use-single-line-haddock-comments.yaml:21]:
```suggestion
  The [Haskell Language Server](https://haskell-language-server.readthedocs.io)
  can directly execute doctest comments, but only if they are single-line.
```

File: decision-log/2023-01-12-use-single-line-haddock-comments.yaml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+decision: >
+  We will use single-line comments so that examples can be executed by HLS and
+  recognised as doctests
+consequences: >
```
Comment (AlexeiDrake) [inline decision-log/2023-01-12-use-single-line-haddock-comments.yaml:25]: I would also mention running doctests as part of the test suite.

**Review by AlexeiDrake — DISMISSED** [review]

**Review by AlexeiDrake — APPROVED** [review]

**Pull request merged**